lib/ui/window.dart

```diff
@@ -143,14 +155,229 @@ class Locale {
   /// ```
   ///
   /// The primary language subtag must not be null. The region subtag is
-  /// optional.
-  ///
-  /// The values are _case sensitive_, and should match the case of the relevant
-  /// subtags in the [IANA Language Subtag
-  /// Registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry).
-  /// Typically this means the primary language subtag should be lowercase and
-  /// the region subtag should be uppercase.
-  const Locale(this._languageCode, [ this._countryCode ]) : assert(_languageCode != null);
+  /// optional. The values are _case sensitive_, and must match the values
+  /// listed in
+  /// http://unicode.org/repos/cldr/tags/latest/common/validity/language.xml and
+  /// http://unicode.org/repos/cldr/tags/latest/common/validity/region.xml.
+  ///
+  /// This method only produces standards-compliant instances if valid language
+  /// and country codes are provided. Deprecated subtags will be replaced, but
+  /// incorrectly cased strings are not corrected.
+  const Locale(
+    this._languageCode, [
+    this._countryCode,
+  ])  : assert(_languageCode != null),
+        this._scriptCode = null,
+        this._variants = null,
+        this._extensions = null;
+
+  /// Creates a new Locale object with the specified parts.
+  ///
+  /// This is for internal use only. All fields must already be normalized, must
+  /// already be canonicalized. This method does not modify parameters in any
+  /// way or do any syntax checking.
+  ///
+  /// * language, script and region must be in canonical form.
+  /// * iterating over variants must provide variants in alphabetical order.
+  /// * The extensions map must contain only valid key/value pairs. "u" and "t"
+  ///   keys must be present, with an empty string as value, if there are any
+  ///   subtags for those singletons.
+  Locale._internal(
+    String language, {
+    String script,
+    String region,
+    collection.SplayTreeSet<String> variants,
+    collection.SplayTreeMap<String, String> extensions,
+  }) : assert(language != null),
+       assert(language.length >= 2),
+       assert(language.length <= 8),
+       assert(language.length != 4),
+       assert((script ?? "Xxxx").length == 4),
+       assert((region ?? "XX").length >= 2),
+       assert((region ?? "XX").length <= 3),
+       _languageCode = language,
+       _scriptCode = script,
+       _countryCode = region,
+       _variants = collection.LinkedHashSet<String>.from(variants ?? []),
+       _extensions = collection.LinkedHashMap<String, String>.from(extensions ?? {});
+
+  /// Parses [Unicode Locale
+  /// Identifiers](https://www.unicode.org/reports/tr35/#Identifiers).
+  ///
```

Comment:
doc nit: Would maybe be more concise as "Parses Unicode Locale Identifiers, which are a subset of BCP 47."

Reply:
It's not a strict subset though, it is also more permissive, e.g. allowing underscores for (non-BCP 47) backward.